code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -1233,6 +1233,19 @@
 		light(user,I)
 		update_brightness(user)
 		return
+	else if(istype(I, /obj/item/weapon/reagent_containers/food/snacks/donut))
+		var/obj/item/weapon/reagent_containers/food/snacks/donut/D = I
+		var/probability = 15*D.soggy
+		to_chat(user, "<span class='notice'>You dip \the [D] into \the [src]</span>")
+		if(prob(probability))
+			to_chat(user, "<span class='danger'>\The [D] breaks off into \the [src]!</span>")
+			D.reagents.trans_to(src,D.reagents.maximum_volume)
+			qdel(D)
+			return
+		reagents.trans_to(D, rand(3,12))
```

Comment:
It seems to. Just tested it with a glass of 10u absinth, once it was empty, it returned to being just a regular glass